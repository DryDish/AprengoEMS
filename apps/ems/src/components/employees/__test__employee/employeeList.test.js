import { render, cleanup } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';

import '@testing-library/jest-dom/extend-expect.js';
import renderer from 'react-test-renderer';
import EmployeeList from "../employeeList.component";

afterEach(cleanup);

it('Renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<EmployeeList />, div);
});

it('Matches the snapshot', () => {
   const employeeListSnapshot = renderer.create(<EmployeeList/>).toJSON();
   expect(employeeListSnapshot).toMatchSnapshot();
});

it('Renders Employee List correctly', () => {
   const {getByTestId} = render(<EmployeeList/>);
   expect(getByTestId("employeeList"))
       .toHaveTextContent("Employee List"
                                && "ID"
                                && "Full Name"
                                && "Phone Number"
                                && "Hours");

});