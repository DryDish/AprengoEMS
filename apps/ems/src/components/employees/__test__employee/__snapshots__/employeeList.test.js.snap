// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<div
  className="col-"
  data-testid="employeeList"
  style={
    Object {
      "margin": "auto",
      "width": "80%",
    }
  }
>
  <h3
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Employee List
  </h3>
  <br />
  <div
    className="container"
  >
    <div
      className="row"
      style={
        Object {
          "fontWeight": "Bold",
        }
      }
    >
      <div
        className="col-1"
      >
        ID
      </div>
      <div
        className="col"
      >
        Full Name
      </div>
      <div
        className="col"
      >
        Phone Number
      </div>
      <div
        className="col-2"
      >
        Hours
      </div>
    </div>
  </div>
  <hr />
  <ul
    className="list-group"
  />
</div>
`;
